Tests/SonarScanner.MSBuild.Common.Test/OperatingSystemProviderTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NSubstitute;
+using static SonarScanner.MSBuild.Common.OperatingSystemProvider;
+
+namespace SonarScanner.MSBuild.Common.Test;
+
+[TestClass]
+public class OperatingSystemProviderTests
+{
+    [TestMethod]
+    public void GetFolderPath_WithUserProfile()
+    {
+        var sut = new OperatingSystemProvider(FileWrapper.Instance);
+        sut.GetFolderPath(Environment.SpecialFolder.UserProfile, Environment.SpecialFolderOption.None)
+            .Should().Be(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile, Environment.SpecialFolderOption.None));
+    }
+
+    [TestMethod]
+    public void DirectoryExists_WithCurrentDirectory()
+    {
+        var sut = new OperatingSystemProvider(FileWrapper.Instance);
+        sut.DirectoryExists(Environment.CurrentDirectory).Should().BeTrue();
+    }
+
+    [DataTestMethod]
+    [DataRow("/etc/os-release", "alpine", true)]
+    [DataRow("/etc/os-release", "ubuntu", false)]
+    [DataRow("/usr/lib/os-release", "alpine", true)]
+    [DataRow("/usr/lib/os-release", "ubuntu", false)]
+    [DataRow("/etc/other-file", "alpine", false)]
```

Comment:
Some distros use double quotes, like e.g. `"rocky"`:
https://github.com/which-distro/os-release/blob/main/rocky/8.6
or
https://github.com/chef/os_release/blob/main/redhat_8

and also the file is supposed to use shell quoting:
> Note that even though this string may not include characters that require shell quoting, quoting may nevertheless be used.
[Source](https://manpages.debian.org/testing/systemd/os-release.5.en.html#:~:text=Note%20that%20even%20though%20this%20string%20may%20not%20include%20characters%20that%20require%20shell%20quoting%2C%20quoting%20may%20nevertheless%20be%20used.)

I do not think this matters for `alpine` but it might be worth adding another os-release file, where 
* ID is not the second entry
* We have double quotes.

I think the Redhat file is a good choice.